public/javascripts/janus.js

```diff
@@ -279,4 +279,16 @@ jQuery(function($){
         }
     });
 
+    // auto-logout (preference persisted via cookie, so server-side can see it when redirecting to federation endpoint)
+    $("#auto_logout_switch").each(function(_, autoLogoutSwitchElement){
+        const COOKIE__AUTO_LOGOUT = "autoLogout"
+        autoLogoutSwitchElement.checked =
+          !!decodeURIComponent(document.cookie)
+          .split(";")
+          .find(_ => _.trim().startsWith(`${COOKIE__AUTO_LOGOUT}=true`));
+        autoLogoutSwitchElement.onchange = (event) => {
+            document.cookie = `${COOKIE__AUTO_LOGOUT}=${event.target.checked}; expires=Fri, 31 Dec 9999 23:59:59 GMT; path=/`
```

Comment:
Would it be simpler to add a parameter to the `consoleLogin` action, rather than have us synchronise the reading and writing of this cookie? We could let the UI choose the most effective way to collect and persist the user's preference, and have that be communicated explicitly when it talks to the server?

In this case, that might mean sticking to localStorage for the preference and including a querystring parameter or optional path parameter to request a "logout-first" console URL?

Side note: The "favourites" feature does use a cookie so there is some prioir art here! However, this is because the favourites feature needs to change the way the homepage renders on the server. The favourites feature is inherently a piece of server-state, so the choice there is infra-side state via a database or to have that state be client-persisted-server-available via a cookie.

Reply:
In the replacement PR I've decided not to take the approach of switching away from cookies. The key reason is that I think we should respect this setting for existing URLs (e.g. those that are bookmarked). This is particularly obvious with the "console url" feature. This feature as-written would have hard-coded the logout decision in the provided URL, which may not be what users expect.